src/engine/controls/cuecontrol.cpp

```diff
@@ -1845,9 +1858,19 @@ void HotcueControl::slotHotcuePositionChanged(double newPosition) {
     emit hotcuePositionChanged(this, newPosition);
 }
 
-void HotcueControl::slotHotcueColorChanged(double newColorId) {
-    m_pCue->setColor(Color::kPredefinedColorsSet.predefinedColorFromId(newColorId));
-    emit hotcueColorChanged(this, newColorId);
+void HotcueControl::slotHotcueColorChanged(double newColor) {
+    if (!m_pCue) {
+        return;
+    }
+
+    mixxx::RgbColor::optional_t color = doubleToRgbColor(newColor);
+    if (!color) {
+        qWarning() << "slotHotcueColorChanged got invalid value:" << newColor;
+        return;
```

Comment:
Due to this return, the co and the cue object are out of sync. 
We can avoid it by connecting the value change request instead.

Reply:
Ok

Reply:
Done.

Reply:
I actually undid that change after some local tests. This does not work properly, e.g. the color is not unset if the color is deleted. Also, it's a bit unsettling to not check if `*color` will crash ;-)

And no other CO in that file checks the value before setting it, hotcue color handling would the be only CO that does this. We can introduce more value change requests to reduce the possible values in the slots, but lets please do that separately, not as part of this PR.